File: src/main/java/io/vertx/ext/sql/UpdateResult.java
package io.vertx.ext.sql;

import io.vertx.codegen.annotations.VertxGen;
import io.vertx.core.json.JsonArray;
import io.vertx.ext.sql.impl.UpdateResultImpl;

/**
 * Represents the result of an update/insert/delete operation on the database.
 * <p>
 * The number of rows updated is available with {@link io.vertx.ext.sql.UpdateResult#getUpdated} and any generated
 * keys are available with {@link io.vertx.ext.sql.UpdateResult#getKeys}.
 *
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
@VertxGen
public interface UpdateResult {

  static UpdateResult create(int updated, JsonArray keys) {
    return new UpdateResultImpl(updated, keys);
  }

  @Deprecated
  int getUpdated();

  /**
   * Get the number of rows updated
   *
   * @return number of rows updated
   */
  int updated();

  @Deprecated
  JsonArray getKeys();

  /**
   * Get any generated keys
   *
   * @return generated keys
   */
  JsonArray keys();

}

File: src/main/kotlin/io/vertx/kotlin/ext/sql/UpdateResult.kt
package io.vertx.kotlin.ext.sql

import io.vertx.ext.sql.UpdateResult

fun UpdateResult(
    keys: io.vertx.core.json.JsonArray? = null,
  updated: Int? = null): UpdateResult = io.vertx.ext.sql.UpdateResult().apply {

  if (keys != null) {
    this.keys = keys
  }

  if (updated != null) {
    this.updated = updated
  }

}

